### PR TITLE
gcalcli: enable tests

### PR DESCRIPTION
**Pull request by AndrewKvalheim**
## Description of changes

~~I noticed that #326353 doesn’t happen when `PYTHONPATH` is unset, though I don’t really understand what the problem was. Is this a reasonable solution, or is there a more conventional way to resolve this?~~

Just [keeping](https://github.com/NixOS/nixpkgs/pull/336883#issuecomment-2307973468) the maintenance commits from this.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`.
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by dotlambda** [comment]
> I noticed that #326353 doesn’t happen when `PYTHONPATH` is unset, though I don’t really understand what the problem was. Is this a reasonable solution, or is there a more conventional way to resolve this?

Your `$PYTHONPATH` shouldn't be set. If it is you have to expect that things break because you're mixing two ways of installing Python modules.

**Comment by AndrewKvalheim** [comment]
> Your `$PYTHONPATH` shouldn't be set.

```console
$ echo ${PYTHONPATH:?It’s not set.}
bash: PYTHONPATH: It’s not set.
```

**Comment by dotlambda** [comment]
> > Your `$PYTHONPATH` shouldn't be set.
> 
> ```
> $ echo ${PYTHONPATH:?It’s not set.}
> bash: PYTHONPATH: It’s not set.
> ```

But `nix-shell` sets `$PYTHONPATH`.

**Comment by AndrewKvalheim** [comment]
This is a [bug in the Nix ecosystem](https://github.com/NixOS/nixpkgs/issues/326353#issuecomment-2307913215) that should be resolved when `nix run` is stabilized.

File: pkgs/applications/misc/gcalcli/default.nix [hunk 1]
```diff
@@ -1,5 +1,10 @@
-{ stdenv, lib, fetchFromGitHub, python3
-, libnotify ? null }:
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  python3,
+  libnotify ? null,
```
Comment (dotlambda) [inline pkgs/applications/misc/gcalcli/default.nix:6]:
```suggestion
  libnotify
```

File: pkgs/applications/misc/gcalcli/default.nix [hunk 2]
```diff
@@ -8,9 +13,9 @@ buildPythonApplication rec {
   version = "4.3.0";
 
   src = fetchFromGitHub {
-    owner  = "insanum";
-    repo   = pname;
-    rev    = "v${version}";
+    owner = "insanum";
+    repo = pname;
```
Comment (dotlambda) [inline pkgs/applications/misc/gcalcli/default.nix:17]:
```suggestion
    repo = "gcalcli";
```
#277994

File: pkgs/applications/misc/gcalcli/default.nix [hunk 3]
```diff
@@ -8,9 +13,9 @@ buildPythonApplication rec {
...
+    rev = "v${version}";
```
Comment (dotlambda) [inline pkgs/applications/misc/gcalcli/default.nix:18]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/applications/misc/gcalcli/default.nix [hunk 4]
```diff
@@ -8,9 +13,9 @@ buildPythonApplication rec {
...
     sha256 = "0s5fhcmz3n0dwh3vkqr4aigi59q43v03ch5jhh6v75149icwr0df";
```
Comment (dotlambda) [inline pkgs/applications/misc/gcalcli/default.nix:19]: Let's use `hash` instead.

File: pkgs/applications/misc/gcalcli/default.nix [hunk 5]
```diff
@@ -1,5 +1,10 @@
...
-, libnotify ? null }:
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  python3,
+  libnotify ? null,
+}:
 
 with python3.pkgs;
```
Comment (dotlambda) [inline pkgs/applications/misc/gcalcli/default.nix:9]:
```suggestion
```

File: pkgs/applications/misc/gcalcli/default.nix [hunk 6]
```diff
@@ -20,13 +25,19 @@ buildPythonApplication rec {
   '';
 
   propagatedBuildInputs = [
```
Comment (dotlambda) [inline pkgs/applications/misc/gcalcli/default.nix:27]:
```suggestion
  dependencies = with python3.pkgs; [
```

File: pkgs/applications/misc/gcalcli/default.nix [hunk 7]
```diff
@@ -20,13 +25,19 @@ buildPythonApplication rec {
...
+    vobject
+    google-api-python-client
+    oauth2client
+    uritemplate
     libnotify
   ];
 
-  # There are no tests as of 4.0.0a4
-  doCheck = false;
+  nativeCheckInputs = [ pytestCheckHook ];
```
Comment (dotlambda) [inline pkgs/applications/misc/gcalcli/default.nix:40]:
```suggestion
  nativeCheckInputs = with python3.pkgs; [ pytestCheckHook ];
```

**Review by dotlambda — COMMENTED** [review]
Let's set `pyproject = true` and add `build-system = [ setuptools ]`.

Also `postPatch` could use `--replace-fail` and `lib.getExe`.

File: pkgs/applications/misc/gcalcli/default.nix (hunk 5 above)
Comment (AndrewKvalheim) [reply pkgs/applications/misc/gcalcli/default.nix]: Thanks for all the review! I noticed other packages using `python3Packages` instead; is that preferred?

**Review by dotlambda — APPROVED** [review]

File: pkgs/applications/misc/gcalcli/default.nix (hunk 5 above)
Comment (dotlambda) [reply pkgs/applications/misc/gcalcli/default.nix]: It's the only thing that supports cross compilation.

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 336883` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>gcalcli</li>
    <li>gcalcli.dist</li>
  </ul>
</details>

**Pull request merged**